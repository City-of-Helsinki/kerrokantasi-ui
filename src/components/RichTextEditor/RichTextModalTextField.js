import React from 'react';
import PropTypes from 'prop-types';

function RichTextModalTextField(props) {
  const {name, label, handleInputChange, handleInputBlur, value, isRequired, errorMsg, formName} = props;
  const errorId = `${formName}-input-error-${name}`;

  return (
    <div className="input-container">
      <label htmlFor={`${formName}-${name}`}>{label} {isRequired && '*'}</label>
      <input
        id={`${formName}-${name}`}
        name={name}
        className="form-control"
        onChange={handleInputChange}
        onBlur={handleInputBlur}
        value={value}
        type="text"
        required={isRequired}
        aria-describedby={errorMsg ? errorId : undefined}
      />
      {errorMsg && <p id={errorId} role="alert" className="rich-text-editor-form-input-error">{errorMsg}</p>}
    </div>
  );
}

RichTextModalTextField.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  handleInputChange: PropTypes.func.isRequired,
  handleInputBlur: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
  isRequired: PropTypes.bool,
  errorMsg: PropTypes.string,
  formName: PropTypes.string,
};

export default RichTextModalTextField;
