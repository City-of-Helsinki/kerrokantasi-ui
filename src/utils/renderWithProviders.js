/* eslint react/prop-types: 0 */
import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { LoginProvider } from 'hds-react';
import { IntlProvider } from 'react-intl';

import createStore from "../createStore";
import { userOidcConfig, apiTokenClientConfig } from './oidcConfig';

const loginProviderConfig = {
  userManagerSettings: userOidcConfig,
  apiTokensClientSettings: apiTokenClientConfig,
  sessionPollerSettings: { pollIntervalInMs: 10000 },
}

const renderWithProviders = (
  ui,
  { preloadedState = {}, locale = 'fi', history, store = createStore(preloadedState), ...renderOptions } = {},
) => {
  const Wrapper = ({ children }) => (
    <LoginProvider {...loginProviderConfig}>
      <IntlProvider locale={locale}>
        <Provider store={store}>{children}</Provider>
      </IntlProvider>
    </LoginProvider>
  );
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};


export default renderWithProviders;
