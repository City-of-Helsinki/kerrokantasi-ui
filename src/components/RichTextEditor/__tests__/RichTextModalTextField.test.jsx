import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import RichTextModalTextField from '../RichTextModalTextField';
import renderWithProviders from '../../../utils/renderWithProviders';

const renderComponent = (propOverrides) => {
  const props = {
    name: 'test-name',
    label: 'test-label',
    handleInputChange: jest.fn(),
    handleInputBlur: jest.fn(),
    value: 'test-value',
    formName: 'test-form',
    isRequired: true,
    ...propOverrides,
  };

  return renderWithProviders(
    <MemoryRouter>
      <RichTextModalTextField {...props} />
    </MemoryRouter>,
  );
};

describe('<RichTextModalTextField />', () => {
  it('should render correctly', () => {
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });

  it('should handleInputChange', async () => {
    const handleInputChangeMock = jest.fn();
    const handleInputBlurMock = jest.fn();

    renderComponent({ handleInputChange: handleInputChangeMock, handleInputBlur: handleInputBlurMock });

    const input = await screen.findByTestId('test-form-test-name');

    const user = userEvent.setup();

    user.type(input, 'test');

    await waitFor(() => expect(handleInputChangeMock).toHaveBeenCalled());

    user.click(document.body);

    await waitFor(() => expect(handleInputBlurMock).toHaveBeenCalled());
  });
});
