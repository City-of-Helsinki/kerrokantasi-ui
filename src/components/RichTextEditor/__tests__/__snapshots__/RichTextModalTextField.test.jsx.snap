// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RichTextModalTextField /> should render correctly 1`] = `
<div>
  <div
    class="input-container"
  >
    <label
      for="test-form-test-name"
    >
      test-label
       
      *
    </label>
    <input
      class="form-control"
      data-testid="test-form-test-name"
      id="test-form-test-name"
      name="test-name"
      required=""
      type="text"
      value="test-value"
    />
  </div>
</div>
`;
